add autofish.lua (#557)
DFHack/scripts: docs/autofish.rst

autofish
========

> .. dfhack_tool::

Reviewer: ```suggestion
.. dfhack-tool::
```

>     :summary: Keep a specific quantity of fish in your fortress.

Reviewer: ```suggestion
    :summary: Auto-manage fishing labors to control your stock of fish.
```

>     :tags: fort auto fps jobs

Reviewer: ```suggestion
    :tags: fort auto fps labors
```

Reviewer: right? it's affecting labor assignment, not the actual job

Author: That's right.

This script makes managing how much fish you keep around automatic. It tries to
maintain a configured stock level of raw and/or prepared fish, partially to keep
item quantities from ballooning out of control, and partly to try and prevent
collecting too many rotten fish.

Usage
-----
``enable autofish``
    Enable the script
``disable autofish``
    Disable the script
``autofish status``
    Show the current status of the script, your configured values, and whether
    or not fishing is currently enabled.
``autofish <max> [min] [<options>]``
    Change autofish settings.

    ``max`` (default: 100) controls the maximum amount of fish you  want to keep
    on hand in your fortress. Fishing will be disabled when the amount of fish
    goes above this value.

    ``min`` (default: 50) controls the minimum fish you want before restarting
    fishing.

    Use ``--include-raw``(``-r``) (default: off) to toggle letting the script
    also count your raw fish as part of your quota. Use it a second time to

>     disable this.

Reviewer: Maybe this should default to on (and be renamed `--toggle-raw`). Isn't it akin to designating trees in autochop? we stop designating there when the expected yield will cause your stock to reach the threshold. Isn't it the same here? otherwise if you have lazy fish cleaners, you'll just keep on amassing raw fish.

Examples
--------

> ``autofish -r 150``

Reviewer: I think the first example should just be `enable autofish`, since that is the simplest use case, and is all a user really needs to do.

autofish
========

.. dfhack-tool::
    :summary: Keep a specific quantity of fish in your fortress.

>     :tags: fort auto fps labor

Reviewer: ```suggestion
    :tags: fort auto fps labors
```
